src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -2576,6 +2596,45 @@ static void start_retriable_subchannel_batches(void* arg, grpc_error* ignored) {
 // LB pick
 //
 
+// The callback to intercept trailing metadata if retries is not enabled
+static void recv_trailing_metadata_ready_for_lb(void* arg, grpc_error* error) {
+  grpc_call_element* elem = static_cast<grpc_call_element*>(arg);
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  if (calld->pick.recv_trailing_metadata != nullptr) {
+    *calld->pick.recv_trailing_metadata =
+        calld->recv_trailing_metadata_op_batch->payload
+            ->recv_trailing_metadata.recv_trailing_metadata;
+  }
+  GRPC_CLOSURE_SCHED(
+      calld->pick.recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+  GRPC_CLOSURE_SCHED(
+      calld->original_recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+  GRPC_ERROR_UNREF(error);
+}
+
+// If needed, intercepts the recv_trailing_metadata_ready callback to return
+// trailing metadata to the LB policy.
+static void maybe_intercept_trailing_metadata_for_lb(
+    grpc_call_element* elem, grpc_transport_stream_op_batch* batch) {
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  if (!batch->recv_trailing_metadata) {
+    return;
+  }
+  if (calld->pick.recv_trailing_metadata_ready != nullptr) {
+    calld->recv_trailing_metadata_op_batch = batch;
+    GRPC_CLOSURE_INIT(&calld->recv_trailing_metadata_ready_for_lb,
+                      recv_trailing_metadata_ready_for_lb,
+                      elem,
+                      grpc_schedule_on_exec_ctx);
```

Comment:
I know we have discussed using `grpc_combiner_scheduler(chand->combiner)` here, but it turns out there's a place that calls this using GRPC_CLOSURE_RUN under a different combiner [here](https://github.com/grpc/grpc/blob/08ef3bca1a10cb3ee46aa2037fe412c75407a067/src/core/ext/filters/message_size/message_size_filter.cc#L185). `grpc_schedule_on_exec_ctx` seems to work. Maybe I am still missing somewhere wrt the execution model.